platform/security/handler/security-handler-pki/src/main/java/org/codice/ddf/security/handler/pki/OcspChecker.java

```diff
@@ -0,0 +1,284 @@
+   * @throws IOException after posting an alert to the admin console, if any error occurs
+   */
+  @VisibleForTesting
+  OCSPReq generateOcspRequest(Certificate cert) throws IOException {
+    try {
+      JcaDigestCalculatorProviderBuilder digestCalculatorProviderBuilder =
+          new JcaDigestCalculatorProviderBuilder();
+      DigestCalculatorProvider digestCalculatorProvider = digestCalculatorProviderBuilder.build();
+      DigestCalculator digestCalculator = digestCalculatorProvider.get(CertificateID.HASH_SHA1);
+
+      CertificateID certId =
+          new CertificateID(
+              digestCalculator, new X509CertificateHolder(cert), cert.getSerialNumber().getValue());
+
+      OCSPReqBuilder ocspReqGenerator = new OCSPReqBuilder();
+      ocspReqGenerator.addRequest(certId);
+      return ocspReqGenerator.build();
+
+    } catch (OCSPException | OperatorCreationException e) {
+      postErrorEvent(
+          "Unable to create a OCSP request. The certificate status could not be verified. "
+              + e.getMessage());
+      throw new IOException();
+    }
+  }
+
+  /**
+   * Sends the {@param ocspReq} request and returns the returned Response
+   *
+   * @param ocspReq - the OCSP request to send
+   * @return the response from the OCSP server
+   * @throws IOException after posting an alert to the admin console, if any error occurs
+   */
+  @VisibleForTesting
+  Response sendOcspRequest(OCSPReq ocspReq) throws IOException {
+    try {
+      SecureCxfClientFactory cxfClientFactory =
+          factory.getSecureCxfClientFactory(ocspServerUrl, WebClient.class);
+      WebClient client =
+          cxfClientFactory
+              .getWebClient()
+              .accept("application/ocsp-response")
+              .type("application/ocsp-request");
+
+      return client.post(ocspReq.getEncoded());
+
+    } catch (ProcessingException | IOException e) {
+      postErrorEvent(
+          "Unable to send OCSP request. The certificate status could not be verified. "
+              + e.getMessage());
+      throw new IOException(e);
+    }
+  }
+
+  /**
+   * Creates a {@link OCSPResp} from the given {@link Response}
+   *
+   * @param response - the response to convert to
+   * @return an OCSP response of the given response
+   * @throws IOException after posting an alert to the admin console, if any error occurs
+   */
+  @VisibleForTesting
+  OCSPResp createOcspResponse(Response response) throws IOException {
+    try {
+      Object entity = response.getEntity();
+
+      if (!(entity instanceof InputStream)) {
+        throw new IOException("Unable to get a response from the OCSP server.");
+      }
+
+      try (InputStream inputStream = (InputStream) entity) {
+        return new OCSPResp(inputStream);
+      }
+    } catch (IOException e) {
+      postErrorEvent(
+          "Unable to send OCSP request. The certificate status could not be verified. "
+              + e.getMessage());
+      throw new IOException(e);
+    }
+  }
```

Comment:
✏️ Alternatively (uses 1 less try block):
```
  OCSPResp createOcspResponse(Response response) throws IOException {
    Object entity = response.getEntity();
    if (!(entity instanceof InputStream)) {
      throw new IOException("Unable to get a response from the OCSP server.");
    }

    try (InputStream inputStream = (InputStream) entity) {
      return new OCSPResp(inputStream);
     } catch (IOException e) {
      postErrorEvent(
          "Unable to send OCSP request. The certificate status could not be verified. "
              + e.getMessage());
      throw new IOException(e);
    }
  }
```

Reply:
The `throw new IOException("Unable to get a response from the OCSP server.");` needs to be caught in the same method first so we can post an alert to the admin console. That's why I have the outer try. I was trying not to add the `postErrorEvent` everywhere but I guess one more won't hurt.